core/src/main/scala/cats/data/NonEmptyList.scala

```diff
@@ -217,6 +219,38 @@ final case class NonEmptyList[+A](head: A, tail: List[A]) {
     NonEmptyList((head, 0), bldr.result)
   }
 
+  /**
+    * Sorts this `NonEmptyList` according to an `Order` on transformed `B` from `A`
+    *
+    * {{{
+    * scala> import cats.data.NonEmptyList
+    * scala> import cats.instances.int._
+    * scala> val nel = NonEmptyList.of(('a', 4), ('z', 1), ('e', 22))
+    * scala> nel.sortBy(_._2)
+    * res0: cats.data.NonEmptyList[(Char, Int)] = NonEmptyList((z,1), (a,4), (e,22))
+    * }}}
+    */
+  def sortBy[B: Order](f: A => B): NonEmptyList[A] = {
+    toList.sortBy(f)(Order[B].toOrdering) match {
+      case x :: xs => NonEmptyList(x, xs)
+      case Nil     => sys.error("absurd")
```

Comment:
This is not really a solution but you can “disable” exhaustivity checking instead:

~~~ scala
match {
  case List(x, xs @ _*) => NonEmptyList(x, xs)
}
~~~

I would add a comment, though, to emphasize that we can not get an empty list since we got our list by converting the `NonEmptyList` into a `List`